spec/azure/virtual_machine.go

```diff
@@ -0,0 +1,20 @@
+package azure
+
+type VirtualMachine struct {
+	// MasterVMSize is the master vm size (e.g. Standard_A1)
+	MasterVMSize string `json:"mastervmsize" yaml:"mastervmsize"`
+	// MasterDataDiskSizeGB is the master vm data disk size in GB
+	MasterDataDiskSizeGB string `json:"masterdatadisksizegb" yaml:"masterdatadisksizegb"`
+	// MasterAdminUsername is the master vm administrator username
+	MasterAdminUsername string `json:"masteradminusername" yaml:"masteradminusername"`
+	//  MasterAdminSSHKeyData is the master vm administrator ssh public key
+	MasterAdminSSHKeyData string `json:"masteradminsshkeydata" yaml:"masteradminsshkeydata"`
+	// MasterOSImagePublisher is the master vm OS image publisher (e.g. GiantSwarm)
+	MasterOSImagePublisher string `json:"masterosimagepublisher" yaml:"masterosimagepublisher"`
```

Comment:
I think we should have an `OSImage` block with Publisher, Offer etc. This will be the same for masters and workers so I think the Master prefix can be removed.

Reply:
See my later comment. I think we should have an OSImage block inside Node.